src/engine/ContextDependentPiecewiseLinearConstraint.cpp

```diff
@@ -0,0 +1,189 @@
+/*********************                                                        */
+/*! \file ContextDependentPiecewiseLinearConstraint.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aleksandar Zeljic
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** A context dependent implementation of ContextDependentPiecewiseLinearConstraint class
+ **/
+
+#include "ContextDependentPiecewiseLinearConstraint.h"
+
+#include "Statistics.h"
+
+ContextDependentPiecewiseLinearConstraint::ContextDependentPiecewiseLinearConstraint()
+   : _numCases( 0 )
+   , _boundManager( nullptr )
+   , _context( nullptr )
+   , _cdConstraintActive( nullptr )
+   , _cdPhaseStatus( nullptr )
+   , _cdInfeasibleCases( nullptr )
+{
+}
+
+ContextDependentPiecewiseLinearConstraint::ContextDependentPiecewiseLinearConstraint(
+    unsigned numCases )
+   : _numCases( numCases )
+   , _boundManager( nullptr )
+   , _context( nullptr )
+   , _cdConstraintActive( nullptr )
+   , _cdPhaseStatus( nullptr )
+   , _cdInfeasibleCases( nullptr )
+{
+}
+
+void ContextDependentPiecewiseLinearConstraint::setActiveConstraint( bool active )
+{
+    if ( nullptr != _cdConstraintActive )
+        *_cdConstraintActive = active;
+    else
+        _constraintActive = active;
+}
+
+bool ContextDependentPiecewiseLinearConstraint::isActive() const
+{
+    if ( nullptr != _cdConstraintActive )
+        return *_cdConstraintActive;
+    else
+        return _constraintActive;
+}
+
+void ContextDependentPiecewiseLinearConstraint::registerBoundManager(
+    BoundManager *boundManager )
+{
+    ASSERT( nullptr == _boundManager );
```

Comment:
we usually compare with the context on the right hand side

Reply:
Fixed